app/decorators/cart_decorator.rb

```diff
@@ -2,8 +2,11 @@ class CartDecorator < Draper::Decorator
   delegate_all
 
   def total_price
-    price = object.cart_items.reduce(0) do |sum,citem| sum + citem.quantity * citem.price end
-    Float("%0.02f" % price)
+    if self.getProp('origin') == 'ncr'
+      self.getProp('amount').to_f
+    else
+      object.cart_items.reduce(0) { |sum,citem| sum + citem.quantity * citem.price }
+    end
```

Comment:
This is a case where having separate decorator classes would be useful for the different origins, but we can tackle that in the Proposal refactoring.